docs/resources/task.md

```diff
@@ -0,0 +1,224 @@
+- `{project}` - Project name; e.g. `ubuntu-os-cloud`.
+- `{family}` - Image architecture; e.g. `ubuntu-2004-lts`.
+
+See https://cloud.google.com/compute/docs/images/os-details for more information.
+
+#### Microsoft Azure
+
+`{user}@{publisher}:{offer}:{sku}:{version}`
+
+Fields:
+
+- `{user}` - User name; e.g. `ubuntu`.
+- `{publisher}` - Image publisher; e.g. `Canonical`.
+- `{offer}` - Image offer; e.g. `UbuntuServer`.
+- `{sku}` - Image SKU; e.g. `18.04-LTS`.
+- `{version}` - Image version; e.g. `latest`.
+
+See https://docs.microsoft.com/en-us/azure/virtual-machines/linux/cli-ps-findimage for more information.
+
+### Kubernetes
+
+- `{image}` - Any [container image](https://kubernetes.io/docs/concepts/containers/images/#image-names).
+
+## Cloud Regions
+
+### Generic
+
+The Iterative Provider offers some common cloud regions which are roughly the same
+for all supported clouds.
+
+- `us-east` - United States of America, East.
+- `us-west` - United States of America, West.
+- `eu-north` - Europe, North.
+- `eu-west` - Europe, West.
+
+### Cloud-specific
+
+In addition to generic regions, it's possible to specify any cloud region
+supported by the underlying cloud provider.
+
+#### Amazon Web Services
+
+- `{region}` - Any [AWS region](https://aws.amazon.com/about-aws/global-infrastructure/regions_az) (e.g. `us-east-1`).
+
+#### Google Cloud Platform
+
+- `{zone}` - Any [GCP **zone**](https://cloud.google.com/compute/docs/regions-zones) (not region) (e.g. `us-east1-a`).
+
+#### Microsoft Azure
+
+- `{region}` - Any [Azure region](https://azure.microsoft.com/en-us/global-infrastructure/geographies) (e.g. `eastus`).
+
+### Kubernetes
+
+The `region` attribute is ignored.
+
+## Known Issues
+
+### Kubernetes
+
+#### Region attribute
+
+Setting the `region` attribute results in undefined behaviour.
+
+#### Directory string format
+
+Unlike public cloud providers, Kubernetes does not offer any portable way of persisting and sharing storage between pods. When specified, the `directory` attribute will create a `PersistentVolumeClaim` with the same lifecycle as the task.
+
+- `{storage_class}:{size}` or
+- `{storage_class}:{size}:{path}`
+
+Fields:
+
+- `{storage_class}` - Name of the storage class; e.g. `local-path`.
+- `{size}` - Size in gigabytes.
+- `{path}` - Local path to synchronize; equivalent to the whole `directory` attribute on public cloud providers. When unspecified, persistent volumes will just be used as a cache and local file synchronization will be disabled.
+
+~> **Warning** Access mode will be `ReadWriteOnce` for `parallelism` equal to 1 or `ReadWriteMany` otherwise.
+
+-> **Note** Rancher's [Local Path Provisioner](https://github.com/rancher/local-path-provisioner) might be the easiest way of deploying a quick `ReadWriteOnce` dynamically allocated storage solution for testing: just run `kubectl apply -f https://raw.githubusercontent.com/rancher/local-path-provisioner/master/deploy/local-path-storage.yaml`
```

Comment:
```suggestion
> **Warning!**: Access mode will be `ReadWriteOnce` for `parallelism` equal
> to 1 or `ReadWriteMany` otherwise.

> **Note**: Rancher's
> [Local Path Provisioner](https://github.com/rancher/local-path-provisioner)
> might be the easiest way of deploying a quick `ReadWriteOnce` dynamically
> allocated storage solution for testing: just run
> `kubectl apply -f https://raw.githubusercontent.com/rancher/local-path-provisioner/master/deploy/local-path-storage.yaml`.
```

Reply:
Do you _really_ want to add a colon immediately after an exclamation mark‽

Reply:
```suggestion
~> **Warning**: Access mode will be `ReadWriteOnce` for `parallelism` equal to 1 or `ReadWriteMany` otherwise.

-> **Note**: Rancher's [Local Path Provisioner](https://github.com/rancher/local-path-provisioner) might be the easiest way of deploying a quick `ReadWriteOnce` dynamically allocated storage solution for testing: just run `kubectl apply -f https://raw.githubusercontent.com/rancher/local-path-provisioner/master/deploy/local-path-storage.yaml`.
```

Reply:
Assuming that `Warning!:` wasn't intentional. If that's not the case, feel free to commit the change.